br/tests/br_pitr/run.sh

```diff
@@ -0,0 +1,105 @@
+#!/bin/bash
+#
+# Copyright 2023 PingCAP, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -eu
+. run_services
+CUR=$(cd `dirname $0`; pwd)
+
+# const value
+WAIT_DONE_CODE=50
+WAIT_NOT_DONE_CODE=51
+PREFIX="pitr_backup" # NOTICE: don't start with 'br' because `restart services` would remove file/directory br*.
+res_file="$TEST_DIR/sql_res.$TEST_NAME.txt"
+
+# start a new cluster
+echo "restart a services"
+restart_services
+
+# prepare the data
+echo "prepare the data"
+run_sql_file $CUR/prepare_data/delete_range.sql
+# ...
+
+# start the log backup task
+echo "start log task"
+run_br --pd $PD_ADDR log start --task-name integration_test -s "local://$TEST_DIR/$PREFIX/log"
+
+# run snapshot backup
+echo "run snapshot backup"
+run_br --pd $PD_ADDR backup full -s "local://$TEST_DIR/$PREFIX/full"
+
+# load the incremental data
+echo "load the incremental data"
+run_sql_file $CUR/incremental_data/delete_range.sql
+# ...
+
+# wait checkpoint advance
+echo "wait checkpoint advance"
+sleep 10
+now_time=$(date "+%Y-%m-%d %H:%M:%S %z")
+echo "get the current time: $now_time"
+export GO_FAILPOINTS="github.com/pingcap/tidb/br/pkg/stream/only-checkpoint-ts-with-check=return(\"$now_time\")"
+set +e # we need to get the exit code of br
+i=0
+while true; do
+    # run br with failpoint to compare the current checkpoint ts with the current time
+    run_br --pd $PD_ADDR log status --task-name integration_test
```

Comment:
injecting a fail point is looks pretty hacky. is it possible to check the status with `jq`?